### PR TITLE
Add more XML comments to Core interfaces

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Add more XML comments to Core interfaces.

### PR Checklist ###

- [x] Targets the correct branch 

#### Does this PR touch anything that might effect accessibility?
No

### CODE REVIEW

File: src/Core/src/Core/ICheckBox.cs [hunk 1]
```diff
@@ -1,7 +1,14 @@
 ﻿namespace Microsoft.Maui
 {
+	/// <summary>
+	/// Represents a View that is a type of button that can either be
```
Comment (hartez) [inline src/Core/src/Core/ICheckBox.cs:4]: A checkbox is not a button. 

Perhaps "Represents a View which allows the user to select a binary choice"?

File: src/Core/src/Core/IGridLayout.cs [hunk 2]
```diff
@@ -2,18 +2,57 @@
 
 namespace Microsoft.Maui
 {
+	/// <summary>
+	/// Represents a layout that arranges views in rows and columns.
+	/// </summary>
 	public interface IGridLayout : ILayout
 	{
+		/// <summary>
+		/// A GridRowDefinition collection for the GridLayout instance.
```
Comment (hartez) [inline src/Core/src/Core/IGridLayout.cs:11]: All of these should probably use the interface names, e.g.:
`An IGridRowDefinition collection for the GridLayout instance.`

**Review by hartez — CHANGES_REQUESTED** [review]

File: src/Core/src/Core/IGridLayout.cs (hunk 2 above)
Comment (jsuarezruiz) [reply src/Core/src/Core/IGridLayout.cs]: Updated.

File: src/Core/src/Core/ICheckBox.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Core/src/Core/ICheckBox.cs]: Yeah, your definitions conveys better what it is.

**Comment by jsuarezruiz** [comment]
@hartez When you can, could you review it again?. Thanks!.

**Review by hartez — APPROVED** [review]

**Pull request merged**